pkg/clusteragent/autoscaling/workload/controller_horizontal.go

```diff
@@ -105,16 +109,18 @@ func (hr *horizontalController) performScaling(ctx context.Context, autoscalerIn
 	scale.Spec.Replicas = hr.computeScaleAction(currentDesiredReplicas, targetDesiredReplicas, minReplicas, maxReplicas, scaleDirection)
 	scaleResult, err := hr.scaler.update(ctx, gr, scale)
 	if err != nil {
-		return nil, fmt.Errorf("failed to update scale subresource for autoscaler %s, err: %w", autoscalerInternal.ID(), err)
+		hr.eventRecorder.Eventf(podAutoscaler, corev1.EventTypeWarning, model.FailedScaleEventReason, "Failed to scale target: %s/%s to %d replicas, err: %v", scale.Namespace, scale.Name, scale.Spec.Replicas, err)
+		return nil, fmt.Errorf("Failed to scale target: %s/%s to %d replicas, err: %w", scale.Namespace, scale.Name, scale.Spec.Replicas, err)
 	}
+	hr.eventRecorder.Eventf(podAutoscaler, corev1.EventTypeNormal, model.SuccessfulScaleEventReason, "Scaled target: %s/%s to %d replicas", scale.Namespace, scale.Name, scale.Spec.Replicas)
```

Comment:
Do we reach this line even if no scaling action was taken (current replicas == desired replicas) ?

Reply:
No it's not reached